handle_actions.go

```diff
@@ -33,68 +33,34 @@ func (m *Mock) DoActions(c *Context) error {
 }
 
 func (m *Mock) doAction(c *Context, a Action) (err error) {
-	var action string
-
+	actualAction := a.GetActualAction()
 	defer func() {
 		if err != nil {
 			logrus.WithFields(logrus.Fields{
 				"err":    err,
-				"action": action,
+				"action": reflect.TypeOf(actualAction).String(),
 			}).Errorf("failed to do action")
 		}
 	}()
-
-	if err := m.doActionSleep(c, a); err != nil {
-		action = "sleep"
-		return err
-	}
-	if err := m.doActionRedis(c, a); err != nil {
-		action = "redis"
-		return err
-	}
-	if err := m.doActionPublishKafka(c, a); err != nil {
-		action = "publish_kafka"
-		return err
-	}
-	if err := m.doActionPublishAMQP(c, a); err != nil {
-		action = "publish_amqp"
-		return err
-	}
-	if err := m.doActionSendHTTP(c, a); err != nil {
-		action = "send_http"
-		return err
-	}
-
-	if err := m.doActionReplyHTTP(c, a); err != nil {
-		action = "reply_http"
-		return err
-	}
-
-	return nil
+	return actualAction.Perform(*c)
 }
 
-func (m *Mock) doActionSendHTTP(c *Context, a Action) error {
-	if structs.IsZero(a.ActionSendHTTP) {
-		return nil
-	}
-
-	sendHTTP := a.ActionSendHTTP
-
-	bodyStr, err := c.Render(sendHTTP.Body)
+func (actionSendHTTP ActionSendHTTP) Perform(context Context) error {
```

Comment:
I'm fine with other naming in the function body, but this receiver `actionSendHTTP` is just too long to read in idiomatic go.

Reply:
you mentioned 'self' was okay?

Reply:
definitely ok with me. the community would pick for example `a` in this case.